packages/ckeditor5-editor-multi-root/src/multirooteditor.ts

```diff
@@ -251,11 +251,11 @@ export default class MultiRootEditor extends DataApiMixin( Editor ) {
 	 * ```
 	 *
 	 * @param rootName Name of the root to add.
-	 * @param isUndoable Whether creating the root can be undone (using the undo feature) or not.
 	 * @param data Initial data for the root.
 	 * @param elementName Element name for the root element in the model. It can be used to set different schema rules for different roots.
+	 * @param isUndoable Whether creating the root can be undone (using the undo feature) or not.
 	 */
-	public addRoot( rootName: string, isUndoable = false, data = '', elementName = '$root' ): void {
+	public addRoot( { rootName, data, elementName }: RootData, isUndoable: boolean ): void {
```

Comment:
I'm not sure if this would not confuse integrators, maybe something like this would better suit our needs especially since only `rootName` is a required parameter.

```suggestion
	public addRoot( rootName, { data, elementName, isUndoable } = {} ): void {
```

Btw you should also update examples in docs.

Reply:
> Btw you should also update examples in docs.

Good point.

> maybe something like this

My proposal focuses on providing root data in one object and non-data flags (behaviors) (`isUndoable` for now, may become an object in next PR) separately.

If anything, I'd go with `rootName, RootData, isUndoable` but i would not move `isUndoable` to the data part. I also make an error, `isUndoable` should still be optional.

Reply:
Okay, I played with it a bit and you are right, I will follow your suggestion.